tools/autograd/derivatives.yaml

```diff
@@ -714,6 +714,10 @@
   self: grad.clone().masked_fill_(self <= other, 0)
   other: grad.clone().masked_fill_(self > other, 0)
 
+- name: fmax(Tensor self, Tensor other) -> Tensor
+  self: grad.clone().masked_fill_(self <= other, 0)
```

Comment:
This is the same formula as for maximum:

https://github.com/pytorch/pytorch/blob/d041f7f2fd46f7f646d959d2ea7c6ca9613d7211/tools/autograd/derivatives.yaml#L714

That seems odd since these are different functions.

Let's look at the truth table:

- if self and other are numbers, then gradient goes to self if self is > other, sure
- if self and other are nan then this comparison is False and self and other both receive gradient, which is interesting
- if either of self or other are nan then this comparison is false and both receive gradient, that seems odd

In particular, if self is a number and other is NaN then it seems like gradient should go to self but not to other? And, symmetrically, if self is NaN and other is a number then it seems like gradient should go to other and not to self?

I think we should also bias towards self receiving grad, all else being equal, and rewrite these expressions as:

```
(self >= other).logical_or_(other.isnan()).logical_not_()
```

Then the masked_fill for other would be:

```
(self >= other).logical_or_(other.isnan())
```

@heitorschueroff Does that look correct to you? What do you think, @Kiyosora?

Reply:
It's does makes sense for me, Thanks for pointing this out, I will fix it soon.

Reply:
Addressed!